crates/notion-core/src/distro/error.rs

```diff
@@ -1,27 +1,69 @@
 //! Provides error types for the installer tools.
 
+use archive::HttpError;
 use notion_fail::{ExitCode, NotionFail};
 
 use failure;
+use reqwest::StatusCode;
+use std::fmt;
+
+// Once Issue #173 is implemented, we can use the ToolSpec struct to differentiate tools
+#[derive(Debug)]
+pub(crate) enum Tool {
+    Node,
+    Yarn,
+}
+
+impl fmt::Display for Tool {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            fmt,
+            "{}",
+            match *self {
+                Tool::Node => "Node",
+                Tool::Yarn => "Yarn",
+            }
+        )
+    }
+}
 
 #[derive(Debug, Fail, NotionFail)]
-#[fail(display = "Failed to download version {} from {}\n{}", version, from_url, error)]
 #[notion_fail(code = "NetworkError")]
-pub(crate) struct DownloadError {
-    version: String,
-    from_url: String,
-    error: String,
+pub(crate) enum DownloadError {
+    NotFound { tool: Tool, version: String },
+    Other { tool: Tool, version: String, from_url: String, error: String },
+}
+
+impl fmt::Display for DownloadError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            DownloadError::NotFound { tool, version } => {
+                write!(fmt, "{} version {} not found", tool, version)
+            }
+            DownloadError::Other { tool, version, from_url, error } => {
+                write!(fmt, "Failed to download {} version {} from {}\n{}", tool, version, from_url, error)
+            }
+        }
+    }
 }
 
 impl DownloadError {
-    pub(crate) fn for_version(
+    pub(crate) fn for_tool_version(
+        tool: Tool,
         version: String,
-        from_url: String,
+        from_url: String
     ) -> impl FnOnce(&failure::Error) -> DownloadError {
-        move |error| DownloadError {
-            version: version,
-            from_url: from_url,
-            error: error.to_string(),
+        move |error| match error.downcast_ref::<HttpError>() {
```

Comment:
I might separate the `match` to a new line and brace the closure body:

```rust
move |error| {
    match error.downcast_ref::<HttpError>() {
        // ...
    }
}
```

But otherwise this looks fine to me.